zkllvm/use-cases/primer.mdx

```diff
@@ -0,0 +1,222 @@
+# Primer
+
+This primer acts as an intro guide to writing complex circuits and compiling them. 
+
+## Dependency management
+
+Most circuits that cover common use cases (e.g., circuits verifying state transitions) require the use of external dependencies. 
+
+In such cases, [**using `clang` or `rustc` directly**](../getting-started/compiling-a-circuit) is discouraged. Instead, it is preferable to use a dedicated build management system:
+
+* CMake for C++
+* Cargo for Rust
+
+:::tip[`clang`]
+
+When calling `clang` directly, dependency management can also be handled via the following methods:
+
+* Via the CLI
+* Via env variables
+* Via updating the system path
+
+Using CMake is still preferable as it offers several valuable features (such as external library detection) that simplify working with complex circuits.
+
+:::
+
+:::info
+
+This primer uses `crypto3` (C++) and `arkworks` (Rust) to show how to use external libraries in a circuit. The primer can be reused for any other suitable library.
+
+:::
+
+### Using CMake
+
+Switch to the working directory:
+
+```bash
+mkdir new_circuit && cd new_circuit
+```
+
+Create a directory for CMake modules:
+
+```bash
+mkdir cmake && cd cmake
+```
+
+Add the [**`CircuitCompile.cmake` module from the zkLLVM repository**](https://github.com/NilFoundation/zkLLVM/blob/master/cmake/CircuitCompile.cmake) to the `./new_circuit/cmake` directory:
+
+```
+new_circuit
+-- cmake
+    -- CircuitCompile.cmake
+```
+
+Create the circuit:
+
+```bash
+cd .. && mkdir src
+```
+
+```bash
+cd src && touch main.cpp
+```
+
+Add circuit code that references an external library:
+
+```cpp
+#include <nil/crypto3/hash/algorithm/hash.hpp>
+#include <nil/crypto3/hash/sha2.hpp>
+
+...
+
+[[circuit]] bool circuit_func() {...}
+```
+
+Create a configuration file for CMake:
+
+```bash
+cd .. && touch CMakeLists.txt
+```
+
+Inside `CMakeLists.txt`, set the project config and include the `CircuitCompile.cmake` module:
+
+```cmake
+cmake_minimum_required(VERSION 3.2)
+
+project(hashes_circuit
+    VERSION 1.0
+    DESCRIPTION "Tutorial circuit with hashes"
+    LANGUAGES CXX)
+
+list(APPEND CMAKE_MODULE_PATH "path/to/cmake")
+
+include(CircuitCompile)
+```
+
+Add the required library to the project configuration:
+
+```cmake
+set(crypto3_DIR "path/to/crypto3_headers")
+
+find_package(crypto3 REQUIRED)
+```
+
+:::tip[Boost]
+
+`crypto3` already depends on Boost. If an alternative library (that does not depend on Boost) is used, add Boost manually:
+
+```cmake
+find_package(Boost REQUIRED COMPONENTS system filesystem)
+```
+
+:::
+
+Set the circuit source and add a build target:
+
+```
+set(SOURCES src/main.cpp)
+
+add_circuit(
+    circuit SOURCES ${SOURCES}
+    LINK_LIBRARIES
+    crypto3::all
+    ${Boost_LIBRARIES}
+)
+
+set(binary_name circuit.ll)
```

Comment:
Setting a variable but not using it after looks a little bit strange. Maybe replace it with a comment: "add_circuit() will generate IR with name circuit.ll"?

Reply:
Yes, you're right. I have amended this part